src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs

```diff
@@ -74,6 +79,22 @@ public bool IsHttp
             }
         }
 
+        /// <summary>
+        /// Whether the source is using the HTTPS protocol.
+        /// </summary>
+        public bool IsHttps
+        {
+            get
+            {
+                if (!_isHttps.HasValue)
+                {
+                    _isHttps = Source.StartsWith("https://", StringComparison.OrdinalIgnoreCase);
```

Comment:
I believe this one should share logic with `IsHttp` since that calculated before this one and does almost same calculation. `StartsWith` is not really heavy method, but still perf matter.

Reply:
Took a stab at it, see if that's what you were thinking.